.ci-builds/.azure-pipelines.yml

```diff
@@ -394,3 +394,64 @@ jobs:
       env:
         CC: clang
         CXX: clang++
+#-------------------------------------------------------------------------
+- job: ICU4C_Cygwin_GCC_x86_64_Release
+  displayName: 'C: Cygwin GCC x86_64 Release'
+  timeoutInMinutes: 120
+  pool:
+    vmImage: 'windows-2019'
+  variables:
+    ICU_CI_CACHE: c:\icu-ci-cache
+    CYG_URL: https://cygwin.com/setup-x86_64.exe
+    CYG_MIRROR: http://mirrors.kernel.org/sourceware/cygwin/
+    CYG_PACKAGES: automake,gcc-core,gcc-g++,make,pkg-config,perl,python3
+    CYG_ROOT: c:\cygwin-root
+    CYG_CACHE: '$(ICU_CI_CACHE)\cygwin64-v3'
+    CYG_CACHED_SETUP: '$(CYG_CACHE)\setup.exe'
+  steps:
+    # Use 'autocrlf input' since checked-in files may already be using CRLF.
+    - script: |
+        git config --global core.autocrlf input
+      displayName: 'Configure Git to checkout with Unix line endings (LF)'
+    - checkout: self
+      lfs: true
+      fetchDepth: 10
+    # Cache expires after 7 days of no activity.
+    - task: Cache@2
+      displayName: 'Restore Cygwin cache'
+      inputs:
+        # Use the contents of the file ".azure-pipelines.yml" as part of the key, as that contains the list of CYG_PACKAGES.
+        # Also include the Cygwin version as part of the key. If we want to use a newer version of Cygwin, we can update the key here.
+        # Note: The version in the key may become out of sync with (older than) the version we actually use if we update this file
+        #       without updating the version in the key. Any updates to this file guarantees that we're using the latest version.
+        key: '"cygwin-3.2" | .ci-builds/.azure-pipelines.yml'
```

Comment:
Just curious, does the cygwin-3.2 need to be enclosed with double quotes? (FWIW, I don't think it really matters, I was just curious if you maybe tried without and needed to add them for some reason perhaps...)

Reply:
Yeah, we need the double quotes here to tell Azure CI that it's a string and not a file path. Otherwise it would error with file not found. The Azure caching docs say 'Keep in mind that any key segment that "looks like a file path" will be treated like a file path', so I guess adding `.ci-builds/.azure-pipelines.yml` as part of the key made it think the unquoted string was a file path.

Reply:
Thanks Daniel.

Hm, in that case I think the Caching docs for Azure pipelines are wrong/misleading.

```
    key: 'yarn | "$(Agent.OS)" | yarn.lock'
```
> In this example, the cache key contains three parts: a static string ("yarn"), the OS the job is running on since this cache is unique per operating system, and the hash of the yarn.lock file that uniquely identifies the set of dependencies in the cache.

I'm going to file an issue on the page for it.

Reply:
Filed an issue here: https://github.com/MicrosoftDocs/azure-devops-docs/issues/10500

Reply:
Actually the docs call out this (I unintentionally excluded the 2nd sentence above):

> This file must exist at the time the task is run. Keep in mind that any key segment that "looks like a file path" will be treated like a file path. In particular, this includes segments containing a `.`.

So since `cygwin-3.2` includes a `.`, it treats it as a file path without the quotes.

Reply:
Hm, I wonder if it is maybe only the `.` that causes it to get messed up then?
(Though in any case, if someone had a file named "yarn" then I think the example would still get messed up.)
I like the idea of using a variable for the version! :)

Reply:
I just tested this with just `cygwin-3.2` as the key without quotes, and Azure CI thinks it's a file -- so it seems it's just the `.` that makes the difference. But good point, filenames without `.` at all would still get messed up. Thanks for filing the issue! :)